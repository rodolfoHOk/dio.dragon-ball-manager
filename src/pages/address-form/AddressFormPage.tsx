export function AddressFormPage() {
  return (
    <div>
      <h1>Formulário de Endereços</h1>
    </div>
  );
}
