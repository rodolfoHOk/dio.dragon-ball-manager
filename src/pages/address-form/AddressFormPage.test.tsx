import { act } from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { AddressFormPage } from './AddressFormPage';

describe('test address form page', () => {
  let fetchMock: jest.MockInstance<Promise<Response>, any> | undefined;

  beforeEach(() => {
    fetchMock = jest.spyOn(global, 'fetch').mockImplementation(() =>
      Promise.resolve({
        ok: true,
        status: 200,
        json: async () => ({
          cep: '01001000',
          bairro: 'Sé',
          logradouro: 'Praça da Sé',
        }),
      } as Response)
    );
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('should render address form', async () => {
    const { container } = render(<AddressFormPage />);

    expect(screen.getByText('Formulário de Endereços')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('CEP')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('bairro')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('logradouro')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should fetch of cep data and fill the form', async () => {
    render(<AddressFormPage />);

    const cepInput = screen.getByPlaceholderText('CEP');
    fireEvent.change(cepInput, { target: { value: '01001000' } });

    await act(() => fetchMock);

    expect(fetchMock).toHaveBeenCalledTimes(1);
    expect(fetchMock).toHaveBeenCalledWith(
      'https://viacep.com.br/ws/01001000/json/'
    );
    expect(screen.getByPlaceholderText('bairro')).toHaveAttribute(
      'value',
      'Sé'
    );
    expect(screen.getByPlaceholderText('logradouro')).toHaveAttribute(
      'value',
      'Praça da Sé'
    );
  });
});
