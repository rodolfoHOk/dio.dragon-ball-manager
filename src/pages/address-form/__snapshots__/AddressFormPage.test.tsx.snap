// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test address form page should render address form 1`] = `
<div>
  <div
    class="flex flex-col gap-4 py-3"
  >
    <h2
      class="font-black text-2xl"
    >
      Formulário de Endereços
    </h2>
    <form
      class="flex flex-row gap-4"
    >
      <fieldset
        class="flex flex-col gap-2"
      >
        <label
          class="font-bold text-lg undefined"
          for="cep"
        >
          CEP
        </label>
        <input
          class="w-full px-4 py-2 flex-1 inline-flex justify-center items-center bg-zinc-100 dark:bg-zinc-900 text-lg font-bold text-zinc-950 dark:text-zinc-50 border border-neutral-500 rounded-md undefined"
          id="cep"
          name="cep"
          placeholder="CEP"
          value=""
        />
      </fieldset>
      <fieldset
        class="flex flex-col gap-2"
      >
        <label
          class="font-bold text-lg undefined"
          for="bairro"
        >
          Bairro
        </label>
        <input
          class="w-full px-4 py-2 flex-1 inline-flex justify-center items-center bg-zinc-100 dark:bg-zinc-900 text-lg font-bold text-zinc-950 dark:text-zinc-50 border border-neutral-500 rounded-md undefined"
          id="bairro"
          name="bairro"
          placeholder="bairro"
          value=""
        />
      </fieldset>
      <fieldset
        class="flex flex-col gap-2"
      >
        <label
          class="font-bold text-lg undefined"
          for="logradouro"
        >
          Logradouro
        </label>
        <input
          class="w-full px-4 py-2 flex-1 inline-flex justify-center items-center bg-zinc-100 dark:bg-zinc-900 text-lg font-bold text-zinc-950 dark:text-zinc-50 border border-neutral-500 rounded-md undefined"
          id="logradouro"
          name="logradouro"
          placeholder="logradouro"
          value=""
        />
      </fieldset>
    </form>
  </div>
</div>
`;
